src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/ListenerPrimary.cs

```diff
@@ -33,6 +33,11 @@ public ListenerPrimary(LibuvTransportContext transportContext) : base(transportC
         {
         }
 
+        /// <summary>
+        /// For testing purposes.
+        /// </summary>
+        public int UvPipeCount => _dispatchPipes.Count;
```

Comment:
That was my first approach...which led to more issues. Setting InternalsVisibleTo on the Transport.Libuv assembly caused conflicts in the test project because both Transport.LibUv and .Core use TaskCache.Sources. This cause compile to fail because Microsoft.Extensions.Internal.TaskCache is ambiguous between the assemblies.